tests/unittests/early_patches.py

```diff
@@ -0,0 +1,20 @@
+import functools
+
+old_lru_cache = functools.lru_cache
+cached_functions = []
+
+
+def wrapped_lru_cache(*args, **kwargs):
+    def wrapper(func):
+        new_func = old_lru_cache(*args, **kwargs)(func)
+        cached_functions.append(new_func)
```

Comment:
nifty registration of `lru_cache` callers.... Not critical, but do we need to check for duplicates before appending. running unittests and dumping cached_functions to a file we get to 47 cached_functions.... yet I only see 17 functions decorated with @lru_cache. Is this because we are importing lru_cached functions in other modules maybe?

Reply:
Good catch. We don't need to check for duplicates as imports only happen once, BUT we were storing any import from anywhere that had an `@lru_cache`, including the stdlib. By the end of tests the number of functions we were storing was over 1000! With my last push, it's now 18 the entire test run.